code/client.dm

```diff
@@ -499,25 +499,14 @@ var/global/list/vpn_ip_checks = list() //assoc list of ip = true or ip = false.
 #endif
 		//Cloud data
 		if (cdn)
-			// Fetch via HTTP from goonhub
-			var/datum/http_request/request = new()
-			request.prepare(RUSTG_HTTP_METHOD_GET, "http://spacebee.goonhub.com/api/cloudsave?list&ckey=[ckey]&api_key=[config.ircbot_api]", "", "")
-			request.begin_async()
-			UNTIL(request.is_complete())
-			var/datum/http_response/response = request.into_response()
-
-			if (response.errored || !response.body)
-				logTheThing("debug", src, null, "failed to have their cloud data loaded: Couldn't reach Goonhub")
-				return
-
-			var/list/ret = json_decode(response.body)
-			if(ret["status"] == "error")
-				logTheThing( "debug", src, null, "failed to have their cloud data loaded: [ret["error"]["error"]]" )
-			else
-				cloudsaves = ret["saves"]
-				clouddata = ret["cdata"]
-				load_antag_tokens()
-				load_persistent_bank()
+			var/tries = 10
+			while(!cloud_available() && tries) // we might sill be waiting for cloud_fetch triggered above
+				tries--
+				sleep(0.1 SECONDS)
```

Comment:
why not use `UNTIL()` for this?

Reply:
Originally I had that there but then if the cloud server is down it'd prevent new player panel from ever appearing so I added an upper limit to the number of iterations.